### PR TITLE
Fix #3189: Create build bazel for each subdirectories of util parser

### LINKED ISSUE #3189: Create a BUILD.bazel file in each of the subdirectories of util/parser
**Describe the solution you'd like**
We want to make each package in util/parser a buildable unit in Bazel.

**Additional context**
There will be references between the packages, so we will have to migrate all the packages together.

### PR DESCRIPTION
**Pull request by yash10019coder**
## Explanation

Fixes #3189 
Created build bazel files for all the subdirs of util/parser
## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by yash10019coder** [comment]
@FareesHussain  @fsharpasharp @BenHenning  getting this error please help

image

**Comment by yash10019coder** [comment]
@anandwana001 please help

File: utility/src/main/java/org/oppia/android/util/parser/html/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        "HtmlParserEntityTypeModule.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        ":dagger",
+        "//third_party:androidx_core_core",
+        "//third_party:androidx_core_core-ktx",
+        "//third_party:javax_inject_javax_inject",
+        "//utility/src/main/java/org/oppia/android/util/logging:console_logger",
+        "//utility/src/main/java/org/oppia/android/util/parser/image:parser",
```
Comment (FareesHussain) [inline utility/src/main/java/org/oppia/android/util/parser/html/BUILD.bazel:38]: This target depends on `//utility/src/main/java/org/oppia/android/util/parser/image:parser`

File: utility/src/main/java/org/oppia/android/util/parser/image/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        "ImageParsingModule.kt",
+        "UrlImageParser.kt",
+    ],
+    visibility = ["//:oppia_api_visibility"],
+    deps = [
+        ":dagger",
+        "//third_party:glide_compiler",
+        "//third_party:javax_inject_javax_inject",
+        "//utility/src/main/java/org/oppia/android/util/logging:console_logger",
+        "//utility/src/main/java/org/oppia/android/util/parser/html:parser",
```
Comment (FareesHussain) [inline utility/src/main/java/org/oppia/android/util/parser/image/BUILD.bazel]: This dependency depends on this target which creates a cycle in the dependency graph.

**Review by FareesHussain — COMMENTED** [review]
@yash10019coder I've left 2 comments which points why you are facing the issue.

**Comment by yash10019coder** [comment]
@FareesHussain 
image
I am getting this error but I have added the deps for glide please help

**Comment by FareesHussain** [comment]
@yash10019coder Add this dependency too
https://github.com/oppia/oppia-android/blob/develop/third_party/versions.bzl#L53

**Comment by yash10019coder** [comment]
@FareesHussain Hi I have moved the tip of my local repo to a earlier commit now how can I do to push this in the remote repo  please help

**Comment by FareesHussain** [comment]
checkout to the latest commit

File: utility/src/main/java/org/oppia/android/util/parser/image/BUILD.bazel (hunk 2 above)
Comment (yash10019coder) [reply utility/src/main/java/org/oppia/android/util/parser/image/BUILD.bazel]: > checkout to the latest commit

but I have committed it and I don't want to merge it with this remote

**Comment by yash10019coder** [comment]
can I delete this remote branch

**Comment by yash10019coder** [comment]
and then make a new PR

**Pull request closed without merging**